onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -374,6 +228,69 @@
             context->GetExecutionInterface(executionObject.GetAddressOf());
             executionObject.As(&commandList);
 
+            auto dftParams = PrepareDFT(context);
+            switch (dftParams.Type)
+            {
+                case DFTType::Stockham:
+                {
+                    StockhamFFT(dftParams, commandList.Get());
+                    break;
+                }
+                case DFTType::BluesteinZChirp:
+                    __fallthrough;
+                default:
+                    return E_NOTIMPL;
+            }
+        }
+        catch (...)
+        {
+            return E_FAIL;
+        }
+
+        return S_OK;
+    }
+
+    DFTParameters PrepareDFT(IMLOperatorKernelContext* context)
+    {
+        DFTParameters params = {};
+
+        // Get the input and output shape sizes
+        ComPtr<IMLOperatorTensor> inputTensor;
+        ORT_THROW_IF_FAILED(context->GetInputTensor(0, &inputTensor));
+        auto inputDims = GetTensorDimensions(inputTensor.Get());
+        auto inputDataSize = ComputeElementCountFromDimensions(inputDims);
+        ML_CHECK_VALID_ARGUMENT(static_cast<size_t>(m_axis) < inputDims.size())
+
+        ComPtr<IMLOperatorTensor> outputTensor;
+        ORT_THROW_IF_FAILED(context->GetOutputTensor(0, &outputTensor));
+        auto outputDims = GetTensorDimensions(outputTensor.Get());
+        auto outputDataSize = ComputeElementCountFromDimensions(outputDims);
+
+        ORT_THROW_HR_IF(E_FAIL, inputDims.size() != outputDims.size());
+
+        // Get optional dft_length input
+        ComPtr<IMLOperatorTensor> dftLengthTensor;
+        if (SUCCEEDED(context->GetInputTensor(1, &dftLengthTensor)) && dftLengthTensor != nullptr)
```

Comment:
>&& dftLengthTensor != nullptr [](http://example.com/codeflow?start=68&length=29)

Can a successful call to `GetInputTensor` result in a `nullptr` output? If so, I'd call that an error per just about every standard COM interface method.

Reply:
This was to handle optional

Reply:
Ok, I wasn't sure if `GetInputTensor` returned `nullptr` and `S_OK` for absent tensors or just a failure code. Next time I touch this area, I'm adding a clear comment to `GetInputTensor` one way or another.